lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -382,6 +431,26 @@ def set_information(self, trans, id, payload={}, **kwd):
             form_values = trans.model.FormValues(user_info_form, user_info_values)
             trans.sa_session.add(form_values)
             user.values = form_values
+
+        # Update values for extra user preference items
+        extra_user_pref_data = dict()
+        get_extra_pref_keys = self._get_extra_user_preferences( trans )
+        if get_extra_pref_keys is not None:
+            for key in get_extra_pref_keys:
+                key_prefix = key + '|'
+                for item in payload:
+                    if item.startswith( key_prefix ):
+                        # Show error message if the required field is empty
+                        if payload[item] == "":
+                            # Raise an exception when a required field is empty while saving the form
+                            keys = item.split("|")
+                            section = get_extra_pref_keys[keys[0]]
+                            for input in section['inputs']:
+                                if input['name'] == keys[1] and input['required']:
+                                    raise MessageException("Please fill the required field")
+                        extra_user_pref_data[ item ] = payload[ item ]
+            user.preferences[ "extra_user_preferences" ] = json.dumps( extra_user_pref_data )
```

Comment:
This is just nitpicking, but should we perhaps rename `extra_user_preferences` to `static_user_preferences` or `admin_defined_user_preferences`, to differentiate this from `dynamic_user_preferences` or `tool_defined_user_preferences` that we may want to implement in the future ?